src/app/ReadClient.cpp

```diff
@@ -1192,6 +1203,10 @@ CHIP_ERROR ReadClient::GetMinEventNumber(const ReadPrepareParams & aReadPrepareP
             // We want to start with the first event _after_ the last one we received.
             aEventMin.SetValue(aEventMin.Value() + 1);
         }
+        else if (aReadPrepareParams.mIsLastEventNumber.HasValue() && aReadPrepareParams.mIsLastEventNumber.Value())
```

Comment:
Please use cluster state cache in chip-tool, where it has recorded the latest event number, or mimic cluster state cache to update the existing callback in chip-tool and record the latest event number.
If you see line above, we try to load the latest event number from callback, and start eventnumber + 1 in next try. 
        ReturnErrorOnFailure(mpCallback.GetHighestReceivedEventNumber(aEventMin));
        if (aEventMin.HasValue())
        {
            // We want to start with the first event _after_ the last one we received.
            aEventMin.SetValue(aEventMin.Value() + 1);
        }
Please don't modify the ReadClient here, thanks

Reply:
Thanks for the review, I did notice GetHighestReceivedEventNumber but it actually functions as ClearValue. And cluster state cache doesn't seem to work for posix chip-tool, I don't know how to modify it, could you provide some guidance on that, thanks!

Reply:
https://github.com/project-chip/connectedhomeip/blob/98bdcd946fb79ccda232335b824cbce6035a53d5/src/app/ClusterStateCache.cpp#L198
the mHighestReceivedEventNumber is updated with the latest number here when we received the event from report, so it is not clearValue, then ReadClient would use next event number to construct event filter based upon  mHighestReceivedEventNumber. one possible integration is to inherit cluster state cache from https://github.com/project-chip/connectedhomeip/blob/98bdcd946fb79ccda232335b824cbce6035a53d5/examples/chip-tool/commands/clusters/ReportCommand.h#L26, this is one reference code in android jni https://github.com/project-chip/connectedhomeip/blob/98bdcd946fb79ccda232335b824cbce6035a53d5/src/controller/java/AndroidCallbacks.h#L85, you could also find the code usage in ios and test code
posix chip-tool support c++ stl, technically it should work when integrating with ClusterStateCache

Reply:
Thanks for the reply, but it seems that this is going to replace the callback, which might be a big change. I've heard that Python runner will replace c++ runner, is this true? Maybe I can remove this part of the change for now, can you review my other changes, thanks!

Reply:
@vivien-apple maybe we should consider to integrate cluster state cache into callback for chip-tool this time, currently android/ios controller both has integrated cluster state cache.

Reply:
I removed the C++ runner changes for now to avoid conflicts with vivien's PR.

Reply:
If it is necessary to add, I can create a new PR.

Reply:
I've removed this part of the change anyway to avoid conflict with vivien's change. If this is really needed, I'll create a new PR.